tracer/src/Datadog.Trace.Coverage.collector/CoverageCollector.cs

```diff
@@ -0,0 +1,253 @@
+                }
+            }
+            else if (testSources is List<string> testSourcesList)
+            {
+                // Process folder
+                foreach (var source in testSourcesList)
+                {
+                    var outputFolder = Path.GetDirectoryName(source);
+                    if (outputFolder is not null)
+                    {
+                        BackupFolder(outputFolder);
+                        ProcessFolder(outputFolder, SearchOption.TopDirectoryOnly);
+                    }
+                }
+            }
+            else
+            {
+                // Process folder
+                BackupFolder(Environment.CurrentDirectory);
+                ProcessFolder(Environment.CurrentDirectory, SearchOption.AllDirectories);
+            }
+        }
+
+        private void OnSessionEnd(object? sender, SessionEndEventArgs e)
+        {
+            _logger?.SetContext(e.Context);
+        }
+
+        private void ProcessFolder(string folder, SearchOption searchOption)
+        {
+            if (_tracerHome is null)
+            {
+                return;
+            }
+
+            var processedDirectories = new HashSet<string>();
+            var numAssemblies = 0;
+            var tracerAssemblyName = typeof(Tracer).Assembly.GetName().Name;
+
+            // Process assemblies in parallel.
+            Parallel.ForEach(
+                Directory.EnumerateFiles(folder, "*.*", searchOption),
+                file =>
+                {
+                    // Skip the Datadog.Trace assembly
+                    if (tracerAssemblyName == Path.GetFileNameWithoutExtension(file))
+                    {
+                        return;
+                    }
+
+                    var extension = Path.GetExtension(file).ToLowerInvariant();
+                    if (extension is ".dll" or ".exe")
+                    {
+                        try
+                        {
+                            var asmProcessor = new AssemblyProcessor(file, _tracerHome, _logger, _ciVisibilitySettings);
+                            asmProcessor.Process();
+
+                            lock (processedDirectories)
+                            {
+                                numAssemblies++;
+                                processedDirectories.Add(Path.GetDirectoryName(file) ?? string.Empty);
+                            }
+                        }
+                        catch (Datadog.Trace.Ci.Coverage.Exceptions.PdbNotFoundException)
+                        {
+                            // If the PDB file was not found, we skip the assembly without throwing error.
+                        }
+                        catch (BadImageFormatException)
+                        {
+                            // If the Assembly has not the correct format (eg. native dll / exe)
+                            // We skip processing the assembly.
```

Comment:
nit: worth debug logging that we skipped it?

Reply:
Changed in: [3257cb2](https://github.com/DataDog/dd-trace-dotnet/pull/2739/commits/3257cb2a2db0bd5fed92b5b64cff1ea50d664468)